### ISSUE TITLE
2.4.0 Alpha

### ISSUE THREAD

**Issue by mackworth** [dscottbuch/cTiVo#63]
Please add a comment here to let me know if you're trying the new version, how extensively, and whether it's working for you or if there's any issue.

**Comment by ITSquire** [comment]
One thing I am noticxing is that in the Router it is using the ISP DNS of
8.8.8.8 will that have an abnormal affect on my TIVO's and other attached
devices?

**Comment by mackworth** [comment]
Nope, that's Google's DNS service; works really well, and shouldn't cause any problem.

**Comment by mackworth** [comment]
Cross posted from Issue #83:

All: Ok, I still can't reproduce the beachball symptom on my system, but I've spent many hours pouring through what might be causing it. Here's a new version (beta 6 (520); without heavy logging) that restructures how I present the download and subscription tables. It's taken a more mainstream approach to some aspects and has cleaned up a couple others, although no Aha! moment that would certainly cause the problem. So please give it a try and let me know if it still has the problem for you.

https://dl.dropboxusercontent.com/u/21507587/cTiVo.zip

Oh, and I should mention that I think I've tested all the fields and buttons, so they should all be hooked up properly, but if you see any other oddities, please let me know.

**Comment by mackworth** [comment]
@ITSquire and @c64syntaxerror   
Check out this article:
http://arstechnica.com/apple/2015/01/why-dns-in-os-x-10-10-is-broken-and-what-you-can-do-to-fix-it/

**Comment by ITSquire** [comment]
Actually it comes up with 192.168.1.20

**Comment by mackworth** [comment]
ok, so try these:
To test this, if you try the following URLs in Safari:
`https://192.168.1.20/TiVoConnect?Command=QueryContainer&Container=%2FNowPlaying&Recurse=Yes&AnchorOffset=0&ItemCount=15 `and 
`https://dvr-213e.local./TiVoConnect?Command=QueryContainer&Container=%2FNowPlaying&Recurse=Yes&AnchorOffset=0&ItemCount=15`

(with account tivo and password equal to your MAK).

If the numbered one works, and the named one doesn't, then that indicates a problem with the DNS. You might try just rebooting your Mac and your router (assuming that's providing the local DNS.

**Comment by czarnik** [comment]
I have tested the 2.4 beta release on both Yosemite & Mavericks. It appears to work well on both. It is running full time on Mavericks in the following config:  Replaced the Decrypted Tivo Show format with a small python script that just creates a link to the downloaded file in the output directory & then uses the batch command to schedule comskip when the system load is light. comskip is run with the nice option to keep system load light and creates the commercial file (*.txt) but leaves the original *mpg file intact. This is done because xbmc / kodi handles playback of the mpg & skips the commercials so no recoding is necessary. Once the txt file is complete, both mpg & txt are moved over to a xbmc content directory via filebot that handles any renaming required and notifies xbmc to rescan the content. Since we don't archive the recorded shows - just watch & auto remove - the larger file size of the mpg isn't an issue and we have access to the high quality audio & video of the HD Tivo recording. Love the product and really applaud the effort. The Tivo content now just 'appears' along with the contents from the internet. Nice work Hugh!

**Comment by WhyAkron** [comment]
Export Subtitles not working in Version 2.4.0b6 (522). Saving as Decrypted TiVo Show, .srt file shows a file size of 3 bytes. When using Version 2.3.1 (468) on the same program, the subtitles export correctly. Started using version 2.4.0b6 for another issue, but haven't used cTivo much since, due to network drive problems (totally unrelated, except for limited drive space to move files), so didn't notice subtitle problem for a while. Also didn't notice (because I usually start a download then go away!) that the progress indication (red bar and stats) doesn't appear for anywhere from a couple of minutes to when the download is almost complete, unless another download is added to the queue, then the indicators appear for the active download. (using OS X 10.9.5 (13F34))

**Comment by mackworth** [comment]
Sorry, ahem, yes, you are correct. Version 2.4.0b6 has a critical bug that interferes with several workflows involving parallel activities (such as subtitles). b7 should be available soon, but I'm still doing some memory testing. I'm not seeing your second problem (in my version 7) but it could be related to the first.

**Comment by mackworth** [comment]
Okay, here's a ["sneak preview" of beta 7 (533)](https://dl.dropboxusercontent.com/u/21507587/cTiVo.zip). I've done a lot of testing, but I'd like to get a couple folks testing it under other environments before designating it. (Yes, I'm a little gun-shy after breaking Handbrake in beta 6).  Versus beta 6, new features are:
- Beta6 bug fix
- Updated to Handbrake version 0.10.0
- Significantly smaller memory usage, especially when running unattended.
- Zero in TiVo Refresh time means manual refresh only. 

Let me know if you've tried it, and that it's working for you.  I <i>don't</i> think this has fixed the Yosemite bug, but it clears the decks, fixing all other known bugs, especially one long-time one that crashed cTiVo whenever mencoder crashed.

**Comment by WhyAkron** [comment]
Tried beta 7 (533) on three programs. Looks like it fixed the 2 issues I was having (no subtitles and delayed progress indicator). I initially thought there were audio and subtitle sync problems, but more likely due to the player I used. (MPlayerX, because QuickTime Player doesn't show captions/subtitles playing Decrypted Tivo Shows format. Sound sync ok in QuickTime Player.)

**Comment by mackworth** [comment]
Great; thanks for the update!  (Have you tried VLC as a player?)

**Comment by tannebil** [comment]
I use VLC all the time on both Windows and OS X. The interface is a mess and it makes a Swiss Army knife look like a masterpiece of simplistic design but it handles everything I throw at it.

**Comment by WhyAkron** [comment]
Had an OLD version of VLC that I didn't like, so updated and tried it again. Like it now. Thanks for the tip. For straight viewing I found Mac Media Player (from Macgo) is decent. Also discovered I have Roxio Video Player (came with Toast 11, but it's kind of hidden and usually triggered by another part of Toast). Has more convenient fast forward/reverse, but otherwise nothing special.

**Comment by mackworth** [comment]
I've posted beta 7 on the [Releases page](https://github.com/dscottbuch/cTiVo/releases). Minor fix included, so it's version 2.4.0beta7 (534) of the program.

**Comment by igregor** [comment]
Greetings.  I just downloaded and installed ctivo 2.4.0b7 on an iMac running OS10.10.2 (Yosemite).  It worked flawlessly to transfer a program from a TiVo Roamio. Nice work.  Thanks.

**Comment by WoodyO** [comment]
All works well except I can't get any subtitles (captions). I am hearing impaired and need this feature. It's probably just me. Using a TIVO Premier and a late model iMac. Please help me.

**Comment by mackworth** [comment]
Sorry you're having a problem. I need a little more info to start helping. First, which cTiVo "Video Format" are you using (some don't have the ability to support subtitles)? Second, when a show is in the download queue, is there a checkmark in the Subtitle column? Third, let me just confirm that you see subtitles while watching TV on TiVo directly? Fourth, which player are you using (Apple's quicktime player)?  Thanks.

**Comment by mackworth** [comment]
Sorry you're having a problem. I need a little more info to start helping. First, which cTiVo "Video Format" are you using (some don't have the ability to support subtitles)? Second, when a show is in the download queue, is there a checkmark in the Subtitle column? Third, let me just confirm that you see subtitles while watching TV on TiVo directly? Fourth, which player are you using (Apple's quicktime player)?  Thanks.

**Comment by mackworth** [comment]
Sorry you're having a problem. I need a little more info to start helping. First, which cTiVo "Video Format" are you using (some don't have the ability to support subtitles)? Second, when a show is in the download queue, is there a checkmark in the Subtitle column? Third, let me just confirm that you see subtitles while watching TV on TiVo directly? Fourth, which player are you using (Apple's quicktime player)?  Thanks.

**Comment by WoodyO** [comment]
Wow! I sure appreciate the quick response. I didn't expect that. I thank you.

To answer your questions: 1) I am using H.264 Medium Quality. 2) Yes, there is a checkmark in both the "skip" and the "Subtitle" columns. 3) Yes, I always use subtitles for everything other than sports whenever watch television (via TIVO). 4) I have tried using both Quicktime Player and Quicktime Player 7.

Thanx again.

WoodyO  ©¿©¬

**Comment by mackworth** [comment]
Great. Two more questions: do you use the Copy to iTunes option, and is there an SRT file in the same folder as your downloaded video (and is there anything in it (you can use a text editor))?  

Then, can you run two experiments for me? First, try using AppleTV format, and assuming that also fails, then try NOT skipping (but still captioning) and see if that works.    

(Sorry for all the questions, but these things are all related due to cTivo's underlying mechanics of moving bits around to the right place at the right time.)

**Comment by WoodyO** [comment]
No, I don't use the Copy option. The SRT files look just fine in listing all the captions (over 900). 

Good news. I switched to Apple TV format and the captions showed up for the first time. But only if I used used VLC as a movie player. "No joy" using Quicktime. I hope we can get it to work w/Quicktime since the transfer is almost twice as fast.

You still want me to try skipping commercials? Ok. I'll do that now.

**Comment by mackworth** [comment]
Actually, the suggestion was to NOT skip commercials, but with the QuickTime format.

As a separate question, try deleting the SRT file and seeing if VLC still shows the captions.

**Comment by WoodyO** [comment]
Sorry I misunderstood. Skipping had no effect other than what it was supposed to.

After deleting the SRT file VLC would no longer display captions.

**Comment by WoodyO** [comment]
Incidentally, The Roxio Player doesn't work any better than Quicktime.

**Comment by WhyAkron** [comment]
Chiming in to add to the mix... Using cTivo 2.4.0beta7 (534) since it was released, OS X 10.9.5 (13F1066). Would say I'm a casual user, but no issues so far. In cTivo I always save as Decrypted Tivo Show to save time. QuickTime Player (Version 10.3 (727.4)) doesn't show captions with or without .srt file. Observation, not complaint - I assume Quicktime needs a different format to show captions. VLC (Version 2.1.5 Rincewind) shows captions with ("Track 1") or without ("Closed captions 1") .srt file. Roxio Video Player (1.0.3 (070)) shows captions with & without .srt file, but appears to not use the .srt.

**Comment by WoodyO** [comment]
Thank you, WhyAkron, for the input. Makes me wonder if OS X 10.10 (Yosemite) might be the culprit.

**Comment by mackworth** [comment]
Ok, so it sounds like cTiVo is creating the subtitles properly (the SRT file), but then isn't embedding the subtitles into the MP4 file for you, which occurs at the end of its processing. (VLC recognizes SRT, but QuickTime Player does not. I just tried that configuration (no commercials, embed captions H.264 Medium quality), and it worked for me.  Can you [download an app called Subler](https://bitbucket.org/galad87/subler/downloads/Subler-0.31.zip), and open the MP4 file and see if it has a subtitle track embedded?

**Comment by mackworth** [comment]
@WhyAkron Correct, the MPG format doesn't have the ability to have subtitles embedded, just the MP4 one, and QuickTime Player does not recognize the external SRT.  On the other hand, because no video recompression occurs with the Decrypted format, the original line 21 subtitle information is still there, so I'm guessing that VLC and Roxio are accessing that (which is lost if any recompression occurs).

**Comment by WoodyO** [comment]
I downloaded Subler.app and opened three cTIVo transfers and all three had subtitle tracks.

**Comment by jerry7171** [comment]
Hello all, since the topic of subtitles has come up I thought I'd ask a question semi-related. I sometimes grab subtitles as I have the odd foreign language show and wanted to keep the subtitles. The problem is I don't know what I'm doing wrong; I have tried exporting the program as a simple decrypted file without encoding since it is much faster on my mid-2011 iMac i5. The problem is I can watch the program in VLC with subtitles but if I decided to convert the file for editing in Final Cut Pro X, I lose the subtitles. I have a vague sense that it is a FCP issue at that point. I also tried exporting directly from my TiVo via different options like using Handbrake or AppleTV but I seem to run into the same thing. As long as I don't molest the file by attempting to edit it it's fine but I lose the subtitles otherwise. Is there a step I'm missing or is this more of an issue outside cTiVo?

**Comment by mackworth** [comment]
Unfortunately, as odd as it may sound, FCP doesn't import subtitles whether embedded or external (just google "FCPX import subtitles"). There are a variety of tools that will help with this, but I haven't experimented with them personally.

**Comment by WoodyO** [comment]
I responded to your last message and am anxious to get this running properly. Is there anything more we can try?

**Comment by mackworth** [comment]
Hi WoodyO. I've been delving into the mysteries of MP4 formats and stss atoms, and I think I found a bug in the subtitle code and have generated a [test version](https://dl.dropboxusercontent.com/u/21507587/cTiVo.zip). See if it helps your situation.

**Comment by WoodyO** [comment]
Ok, we are one step closer. Using your test version of cTIVo I was able to:
     1)  transfer using the H.264 Medium Quality video format and
     2)  See captions using the VLC Player but
     3)  NOT see captions using Quicktime or Quicktime 7 players

Almost there. Are you using Yosemite?

**Comment by mackworth** [comment]
I am indeed, and see no problem with QuickTime Player with H.264 Medium Quality. With Subler, can you Export the subtitle track (from one you've created with the test version) to junk.SRT, and compare it to the SRT file already in the directory (should be exactly the same).  

Can you try copying the file into iTunes and viewing it there (as well as on an iOS device if you have one) to see if it offers the caption option?

BTW, to really confirm the VLC ones, you should move the SRT file OUT of the directory, as VLC seems to prefer the SRT if it's available (and you may then have to explicitly turn on the subtitle display).

**Comment by mackworth** [comment]
Question Woody: if you have QT7 installed, do you also have Perian installed (look in System Preferences)?  If so, try removing it (just click on Perian and hit Remove, then Show All preference panes and Control-Click on Perian to hit Remove again). It hasn't been updated for Yosemite (see http://www.perian.org) , and may be interfering

**Comment by WoodyO** [comment]
I made the comparison of the two .srt files and they are the same visually throughout their lengths. I can't remember how to do a digital comparison. Since I didn't think VLC would run without an .srt file I tried playing using both versions and they both worked so I assume the files are identical.

Captions showed in VLC with or without the srt file w/captions turned on.

I imported a file into iTunes. The movie with imbedded captions again would not show them.

No, do not have Perian installed. Never heard of it. I only kept QT7 because is had some editing capability and seems to run better than more current versions.

Your ball.

**Comment by mackworth** [comment]
Hmm. I'd like to see one of the files; if you have a short one and can post it somewhere (GitHub will only take image files). If you don't have a short one, just manually record a short piece on Tivo, and run it through the process.

**Comment by WoodyO** [comment]
I will create a short file but I don't know how to get it to you. Perhaps another email account? Or via the cloud?

**Comment by mackworth** [comment]
Any cloud account you have that you can point an URL to (e.g. DropBox public folder)

**Comment by WoodyO** [comment]
I have an iCloud Drive but have only used it for sending emails. But I don't know how to create a URL to point to it. Sorry.

Also I am trying to coral my daughter who has a iPod to try to get a sample file run on it. I have no iOS devices.

Rooting for Kentucky! Catch you again on Monday.

**Comment by mackworth** [comment]
Yeah, it looks like icloud drive can only share photos and iwork-type documents. Probably easiest to open a free Dropbox account (here's my referral link: https://db.tt/26vuVBV0 )

**Comment by WoodyO** [comment]
No comment.

![dt150404](https://cloud.githubusercontent.com/assets/11732429/6995740/932c2aa8-db10-11e4-9ba5-804adf619807.gif)

**Comment by dscottbuch** [comment]
If you have Yosemite and use the Apple mail client you can use Mail Drop. 
http://www.tekrevue.com/tip/how-to-send-large-email-attachments-with-mail-drop/


D. Scott Buchanan

Sent from my iPhone

This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email in error please notify the system manager

**Comment by WoodyO** [comment]
Scott,
Thank you. As I understand it the problem here is that Github cannot received email movie attachments. Only graphics.

WoodyO  ©¿©¬

**Comment by mackworth** [comment]
That's correct, but Scott's suggesting that you email the file directly to me, using Apple's Mail Drop.  I think he's correct, so just use Yosemite's mail client to send it to my aol address (which to avoid putting the actual address here to avoid unwanted mail, is the exact same username as here, but with aol  dot-com server after the at-sign). After you hit send, the client will object that your server can't handle that big a file, and ask if you want to use Mail Drop. Just say Yes, and I'll let you know if I got it.

**Comment by dscottbuch** [comment]
Attachment available until May 6, 2015
Just to test if mail drop will send a link through git-hub here’s a 30 MB nonsense file
> On Apr 6, 2015, at 9:26 AM, Hugh Mackworth <notifications@github.com <mailto:notifications@github.com>> wrote:
> 
> That's correct, but Scott's suggesting that you email the file directly to me, using Apple's Mail Drop. I think he's correct, so just use Yosemite's mail client to send it to my aol address (which to avoid putting the actual address here to avoid unwanted mail, is the exact same username as here, but with aol dot-com server after the at-sign). After you hit send, the client will object that your server can't handle that big a file, and ask if you want to use Mail Drop. Just say Yes, and I'll let you know if I got it.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/dscottbuch/cTiVo/issues/63#issuecomment-90132308>.
> 
Click to Download
 <https://www.icloud.com/attachment/?u=https%3A%2F%2Fs3-us-std-101-prod.digitalhub.com%2FB%2FAWmwjtfdRHRbJVxl5zVu5LQ78tvcAch16_q8SkqPkkLiJ5rCWvD-Bsa2%2F%24%7Bf%7D%3Fo%3DAo1Btx6PP8ih60GTxRwO03oYyXWmNrxjQnap5gNW1nHy%26v%3D1%26x%3D3%26a%3DBbPF4u1siNPwA8LiKgEA_wHIAP8jtPQ6%26e%3D1430937708%26k%3D%24%7Buk%7D%26r%3D3D074BBD-E937-4FA9-97F1-535786A569D2-1%26ckc%3Dcom.apple.largeattachment%26ckz%3DBCE114CF-DE73-41B7-92E6-9DA83CC3E954%26z%3Dhttps%253A%252F%252Fp04-content.icloud.com%253A443%26s%3D7KH6WSV_pzQb83z8QHHU9SoemNg&uk=3gAkJ-qG_PBBKOgWKqZKDQ&f=xaa&sz=31457280>xaa
31.5 MB

**Comment by mackworth** [comment]
Hey, look at that!  Scott has shown that you can just reply to a github posting in Mail.app and attach a large file. Apple hosts the file automagically, Github posts the URL, and I can then download it.  (And I've removed my email from the above notes because I get enough spam already.)

**Comment by dscottbuch** [comment]
Just FYI file size limit is 5GB according to apple web site. 

D. Scott Buchanan

Sent from my iPhone

This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email in error please notify the system manager

**Comment by WoodyO** [comment]
I sent nonsense file via iCloud mail but the server didn't recognize the address. I used "notifications" as a name. Did I misunderstand the email address you wanted? What should the name have been? I used iCloud mail account which I know handles "large" messages.

I then tried Scotts suggestion but received the following error message:

![screen shot 2015-04-06 at 10 07 19 pm](https://cloud.githubusercontent.com/assets/11732429/7017509/3dc47f36-dcad-11e4-93b0-ed99cb160908.jpg)

WoodyO  ©¿©¬

**Comment by mackworth** [comment]
Well, the good news is that it's recognizing the file size as being too big, so that's a good start. I just tried it and you do have to enable it per sending account.  Check in Mail>Accounts>{Your icloud account>Advanced for "Send large attachments with Mail Drop". Turn that on  and save the account and you should be set.

**Comment by dscottbuch** [comment]
Interesting.  I didn't do that explicitly. It must have been on by default. 

D. Scott Buchanan

Sent from my iPad

This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed. If you have received this email in error please notify the system manager

**Comment by WoodyO** [comment]
My iCloud account was already checked for "large" files. As I mentioned earlier I have no trouble sending large files using my iCloud mail account. I have shared movies a number of times with others. I think the reason I don't get the error message that Scott gets is because I already had an iCloud account AND an iCloud Drive.

WoodyO  ©¿©¬

**Comment by WoodyO** [comment]
Hugh, if you could give me an email address that CAN accept movie files it should probably solve this dilemma.

**Comment by mackworth** [comment]
Actually the error msg above suggests you were sending from the non-icloud acct astound.  Is this one enabled?  

my email address is the same as this github accounts, but at aol.com

**Comment by WoodyO** [comment]
Yes, that is correct. A non-iCloud account since that is what was requested that we try (Drop Box and so forth). You will recall that when I did use the iCloud account AOL declared that address as invalid. That is one of the reasons I dropped my AOL account: Too many rejected incoming emails that neither I nor AOL could explain. AOL is awfully fussy. However, I'll try again.

**Comment by WoodyO** [comment]
I sent the nonsense file again and again AOL has rejected the address. I am convinced that I am using the wrong AOL address. Since you have my personal email address why don't you just tell me the address I should be using. I promise not to tell anyone.

**Comment by Deputy5211** [comment]
Greetings.

I absolutely LOVE cTiVo.  Sliced bread, watch out!

I am running version Version 2.4.0b7 (534) on Yosemite 10.10.3, and I have just started noticing the (C) copyright symbol in "Status" column.  These shows are greyed out and cannot be downloaded.

This was not the case until quite recently. I have been able to transfer several new episodes of the shows in question until a week or two ago.   

Any thoughts?  Was there a recent TiVo update that now flags these shows and blocks their transfer?  Something in cTiVo that can be adjusted to enable these to pick up again?

**Comment by mackworth** [comment]
Thanks for the compliment!

(C) means that your show has been marked by your TV provider (presumably cable) as "copy-once" . It's called the "CCI byte", which lets content providers mark how much copying is allowed (copy-none will prevent tivo from recording, copy-once lets it record but not retransmit, and copy-freely lets it redistribute). And no, the TiVo will not let us download those shows. (You can show/hide those Protected shows in the Options menu).

It's up to them (Comcast, TW, whomever) to decide, although sometimes they have to do so by contract (e.g. HBO). TW is famous for marking everything, Comcast is pretty open. 

I have no idea why it would have changed, but you can call your cable provider and ask them what changed.

Here's a couple of articles about it:
http://consumerist.com/2012/06/12/time-warners-copy-protection-might-just-drive-tv-lover-to-piracy/
http://zatznotfunny.com/2012-02/the-best-worst-cable-companies-for-tivo-owners/

 I'm pretty sure it's not legal for them to mark OTA content as copy-once,

**Comment by Deputy5211** [comment]
Thank you for the prompt and detailed response. I believe I may have determined the cause. 

Previously, I had the TWC cable plugged directly into the TiVo. When TWC Dallas went all-digital, I hooked up one of their cable boxes.  The dates match up perfectly. 

Analog cable was not carrying the CCI byte in its data, and/or TiVo was not recognizing it. 

The TWC box, however, decoded the byte and passed it in the data stream to the TiVo, which stores it with other program info, and cTiVo displayed it. 

Mystery solved.  

Next test will be to determine if a "dumb" digital to analog converter box (as opposed to TWC's smart Motorola box) will handle the CCI any differently.   

Your thoughts?

Antony
 
--
From Antony's iPhone 6
--

--
This e-mail message, including any attachments, is for the sole use of the intended recipient(s) and may contain information that is confidential and protected by law from unauthorized disclosure. Any unauthorized review, use, disclosure or distribution is prohibited. If you are not the intended recipient, please contact the sender by reply e-mail and destroy all copies of the original message.

**Comment by Deputy5211** [comment]
Hugh,

One request, not related to our existing thread. 

In the middle window, download queue, would be great if the indicators Complete, Downloading, TiVo Deleted, Encoding, <null>, et al, were broken out into a separate display field from the show's title and made sortable.  

This change would allow a user to sort display on this criteria and easily see what was left to process. It is currently a PITA to scroll up and down to see what is in process, what is pending, and so on. 

Best,
Antony

 
--
From Antony's iPhone 6
--

--
This e-mail message, including any attachments, is for the sole use of the intended recipient(s) and may contain information that is confidential and protected by law from unauthorized disclosure. Any unauthorized review, use, disclosure or distribution is prohibited. If you are not the intended recipient, please contact the sender by reply e-mail and destroy all copies of the original message.

**Comment by mackworth** [comment]
@Deputy5211 sorry for delay.  That chronology makes sense, and the lower-end converter might help; although the real answer is to get TWC to change behavior!

Good idea on the separate field; it's not quite as easy as it sounds, as we have to put the download progress field somewhere, so I'll see what I can do.  Please create a separate issue for it though, so I can track it as an enhancement.

**Comment by Deputy5211** [comment]
Roger that. 

 
--
From Antony's iPhone 6
--

--
This e-mail message, including any attachments, is for the sole use of the intended recipient(s) and may contain information that is confidential and protected by law from unauthorized disclosure. Any unauthorized review, use, disclosure or distribution is prohibited. If you are not the intended recipient, please contact the sender by reply e-mail and destroy all copies of the original message.

**Comment by mackworth** [comment]
Well, for those of you following along, I have another shot at fixing the Yosemite issue. Please try Beta8, now posted in the [Releases section](https://github.com/dscottbuch/cTiVo/releases/tag/v2.4.0-Beta8)

**Comment by mackworth** [comment]
And here's beta 9, hopefully the last before a general release...
[Releases section](https://github.com/dscottbuch/cTiVo/releases/tag/v2.4.0-Beta9)

**Comment by tannebil** [comment]
I don’t think anything I do is going to hit changes in a way I can see. I’ll flush my caches and let it reload everything from mytvdb but is there anything specifically you’d like me to try?

**Comment by mackworth** [comment]
Not too concerned about ensuring the rate limiting is working, mostly just looking for any regressions (wrong episode numbers or things like that. Manually downloading season artwork should work now.  Also, see if the movie artwork is working for you. 

Thanks!

**Comment by tannebil** [comment]
OK, I’ll try a movie. How do I manually download season artwork? I’ve never done that and I don’t see it anywhere?

**Comment by mackworth** [comment]
There's nothing in ctivo to get the file, so you can go to imdb or thetvdb. Once you have a file (e.g.Bones S09.jpg) , just put it in <your downloaddir>/thumbnails and it should pick it up for future shows of that season (over downloading the episode-specific one)

**Comment by tannebil** [comment]
What's new in the way it works?

**Comment by mackworth** [comment]
For this part, just the prioritization, season before download. As iTunes seems to use the same image for the whole season, this may make more sense. Now that you ask, I'll have to see if there's a way to find the season images on thetvdb  
H

**Comment by abro-git** [comment]
Hi, I'm having an issue with my downloads failing.  I'm running OS X 10.9.5 (Mavericks, not Yosemite) and cTiVo Version 2.4.0b8 (537).  I have a TiVo Roamio Plus.

I can see my Tivo in the app.  I can select and queue recordings in the app.  They retry several times and then fail.  

I can login to the Tivo directly via my browser using the MAK as the password and download the .tivo file that way.  Downloads are enabled on the Tivo.com.  I've rebooted the Tivo box multiple times.

What other steps should I take toward making this configuration work as anticipated?  Please advise.

Thanks,
Adam

PS  Here's my log file: 
2015-05-29 18:51:28:583 MTAppDelegate>applicationDidFinishLaunching:@169>Starting cTiVo; version: 537
2015-05-29 18:51:30:793 MTTiVo>connection:willSendRequestForAuthenticationChallenge:@762>Family Room challenge failed
2015-05-29 18:51:30:793 MTTiVo>connection:didFailWithError:@790>URL Connection Failed with error Error Domain=NSURLErrorDomain Code=-1012 "The operation couldn’t be completed. (NSURLErrorDomain error -1012.)" UserInfo=0x7fe65bd27d90 {NSErrorFailingURLKey=https://DVR-452C.local./TiVoConnect?Command=QueryContainer&Container=%252FNowPlaying&Recurse=Yes&AnchorOffset=0&ItemCount=15, NSErrorFailingURLStringKey=https://DVR-452C.local./TiVoConnect?Command=QueryContainer&Container=%252FNowPlaying&Recurse=Yes&AnchorOffset=0&ItemCount=15}
2015-05-29 18:55:47:937 MTTiVoManager>updateTiVoDefaults:@487>Warning: didn't find tivo Family Room
2015-05-29 18:59:17:177 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Penny Dreadful: Evil Spirits in Heavenly Places: Can't Download
2015-05-29 19:00:17:196 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Penny Dreadful: Evil Spirits in Heavenly Places: Can't Download
2015-05-29 19:01:17:217 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Penny Dreadful: Evil Spirits in Heavenly Places: Can't Download
2015-05-29 19:02:17:236 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; cancelled./Penny Dreadful: Evil Spirits in Heavenly Places: End Download
2015-05-29 19:03:17:250 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Scandal: A Few Good Women: Can't Download
2015-05-29 19:04:17:269 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Scandal: A Few Good Women: Can't Download
2015-05-29 19:05:17:290 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Scandal: A Few Good Women: Can't Download
2015-05-29 19:06:17:310 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; cancelled./Scandal: A Few Good Women: End Download
2015-05-29 19:11:35:302 MTTiVo>connection:didFailWithError:@790>URL Connection Failed with error Error Domain=NSURLErrorDomain Code=-1001 "The request timed out." UserInfo=0x7fe65bf1dfe0 {NSUnderlyingError=0x7fe65e9558b0 "The request timed out.", NSErrorFailingURLStringKey=https://dvr-452c.local./TiVoConnect?Command=QueryContainer&Container=%252FNowPlaying&Recurse=Yes&AnchorOffset=0&ItemCount=15, NSErrorFailingURLKey=https://dvr-452c.local./TiVoConnect?Command=QueryContainer&Container=%252FNowPlaying&Recurse=Yes&AnchorOffset=0&ItemCount=15, NSLocalizedDescription=The request timed out.}
2015-05-29 19:21:30:182 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Scandal: A Few Good Women: Can't Download
2015-05-29 19:22:30:203 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Scandal: A Few Good Women: Can't Download
2015-05-29 19:23:37:680 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Scandal: A Few Good Women: Can't Download
2015-05-29 19:24:38:670 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; cancelled./Scandal: A Few Good Women: End Download
2015-05-29 19:25:38:690 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Despicable Me: Can't Download
2015-05-29 19:26:38:709 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Despicable Me: Can't Download
2015-05-29 19:27:38:730 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Despicable Me: Can't Download
2015-05-29 19:28:38:751 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; cancelled./Despicable Me: End Download
2015-05-29 23:07:29:722 MTAppDelegate>cleanup@998>cTiVo exiting
2015-05-29 23:07:35:478 MTAppDelegate>applicationDidFinishLaunching:@169>Starting cTiVo; version: 537
2015-05-31 15:23:55:378 MTTiVoManager>netServiceBrowser:didRemoveService:moreComing:@1503>Removing Service: Family Room
2015-05-31 15:24:29:409 MTTiVo>tivoNetworkCallback@172>Tivo Family Room is now offline
2015-05-31 15:24:29:831 MTTiVo>tivoNetworkCallback@172>Tivo Family Room is now online
2015-05-31 15:32:41:334 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Despicable Me: Can't Download
2015-05-31 15:33:41:351 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Despicable Me: Can't Download
2015-05-31 15:34:41:368 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; retrying.../Despicable Me: Can't Download
2015-05-31 15:35:41:386 MTTiVoManager>notifyWithTitle:subTitle:isSticky:forNotification:@1299>Notify: TiVo show failed; cancelled./Despicable Me: End Download

**Comment by tannebil** [comment]
i must be missing something about doing the movie art. Is there a setting because i did three movies and didn’t get any art work?

**Comment by mackworth** [comment]
Nope, provided you've got "Get Artwork from TVDB.com" turned on in Preferences, it should work. If you do, could you set MTTiVoShow to Detail and see if there's any error messages? You can also look in your temporary directory (`Cmd_G /tmp/cTiVo` in the Finder) after a movie starts downloading to see if there's an appropriate jpeg (`MovieTitle_Year.jpg`)

**Comment by WoodyO** [comment]
I have tried Beta 9 and it appears flawless to me. I did note that during the encoding stage it became particularly CPU intensive and seemed to stall when running in the background on my iMac. I'm sure this is to be expected but thought I'd mention it since it runs pretty hot.

WoodyO  ©¿©¬ 

===============================================================

**Comment by mackworth** [comment]
Glad to hear it's flawless! 
Yes, video encoding's about the most CPU intensive activity there is. Does it seem to be more intensive than before v9?

**Comment by WoodyO** [comment]
During the encoding stage - yes. On the other hand the actual downloading appears faster, briefly as high as 6MB/sec!!! These are, of course, spikes but none-the-less impressive. Another thing I noticed was that the commercial skipping was running truer. Or is that my imagination?

===============================================================

**Comment by tannebil** [comment]
That’s why. I don’t have it it turned on since I want to use my own art work for TV shows.  Since the movie art is coming from a different site than theTVDB.com <http://thetvdb.com/>, I didn’t realize it was controlled by the same preference.

**Comment by mackworth** [comment]
Ah. Note that with the season artwork change I mentioned, if your own art is available, it will take precedence over the downloaded. It now looks for episode-specific, then season-specific, then series, then downloaded.

**Comment by tannebil** [comment]
I turned it on, flushed the cache, and it worked fine for a movie. I next tested with a TV show that has already has season artwork downloaded. It ignored the existing season artwork in favor of the episode artwork on thetvdb.com <http://thetvdb.com/>. The episode artwork is usually just a screen grab from the episode without any identifying information. 

iTunes artwork has always been a mystery to me but I thought the season and series artwork is just artwork iTunes grabs from one of the episodes, i.e. there isn’t actually any series/season metadata except what exists in the episode metadata. If all the episodes get the episode artwork, how will series/season artwork ever get into iTunes? I struggled with this a couple of years back and ended up having to put the season artwork in all the episodes to get it to display reliably in season displays but that was a few versions of iTunes back. Even now, when I look at a series display where I have multiple seasons, the artwork displayed for the series is usually from the latest series but sometimes it is for the previous series.

Perhaps you’ve cracked the black box of series and season artwork in iTunes?

**Comment by mackworth** [comment]
'Perhaps you’ve cracked the black box of series and season artwork in iTunes?'
Hardly. Just trying to make the scenario you describe work better: save a season image and cTiVo should apply it to all episodes in that season.  

'there isn’t actually any series/season metadata except what exists in the episode metadata"
Correct. Both iTunes and the iOS devices show the first episode's artwork as the season artwork.  iTunes then shows the embedded episode's artwork for each episode; however, iOS Video doesn't. It picks a random screen shot from the video. 

Where did you put the "existing season artwork" and what was it named? cTiVo should have picked it up.

**Comment by mackworth** [comment]
I have released beta 11. If you missed v10, just as well.  Version 10 fixed a possible v9 crash with enabling scheduling, but introduced another problem, fixed in v11. Except for those glitches, it's looking good for a release, so let me know if you find any problem.

@Deputy5211  I've added a column DL Status per your request.

**Comment by WhyAkron** [comment]
Just downloaded and started using beta 11. Only one thing so far, and it's a personal preference. I was using beta 7, so I don't know when it changed, but I liked it better when the progress/color bar tracked over the program title in the Download Queue column. It was easier to see, being the widest column. But that's just me, not a big deal. Haven't given 11 any kind of workout yet, just a couple of downloads, so far so good. For the record, OSX 10.9.5 on a mid-2010 MacBook Pro.

**Comment by mackworth** [comment]
So, it shows the download progress bar and stage in one of three columns: either DL Stage, Download Queue, or Series in that priority order. So if you just hide the DL Stage column, then the progress bar  and stage will still show in the Download Queue column.

**Comment by WhyAkron** [comment]
OK. Thank you. Also just realized that diownload status now has its own column.

**Comment by Deputy5211** [comment]
Which I think is awesome, so can easily sort and find what is pending, complete, failed, etc, rather than having to scroll the entire list and risk missing. 

 
--
From Antony's iPhone 6
--

--
This e-mail message, including any attachments, is for the sole use of the intended recipient(s) and may contain information that is confidential and protected by law from unauthorized disclosure. Any unauthorized review, use, disclosure or distribution is prohibited. If you are not the intended recipient, please contact the sender by reply e-mail and destroy all copies of the original message.

**Comment by WoodyO** [comment]
Is there a link to download V11?

**Comment by mackworth** [comment]
Sure thing:  [Releases section](https://github.com/dscottbuch/cTiVo/releases/tag/2.4.0-beta11)

**Comment by WoodyO** [comment]
Hugh,

I am beginning to feel like the village idiot. I've clicked on everything in sight on the Releases: 2.0.4 Beta 11 page but don't get anything to download. Am I on the wrong page?

WoodyO  ©¿©¬ 



===============================================================

**Comment by mackworth** [comment]
That's strange. The first link listed `cTiVo.zip`in the `Downloads` section of that page worked just now for me. It should be the same as [this link](https://github.com/dscottbuch/cTiVo/releases/download/2.4.0-beta11/cTiVo.zip). It won't take you anywhere, should just start a download. Where are you geographically?

**Comment by WoodyO** [comment]
Strange indeed. This link worked perfectly. Just for the hell of it I retried the first link again and sure enough it failed exactly as before and took me to that same “Releases" page. I am in the thriving metropolis of Walnut Creek, CA (Northern California).

===============================================================

**Comment by mackworth** [comment]
The first one did point to the releases page, but the ctivo.zip at the bottom should work. Oh well glad you got it!

**Comment by UbiGit** [comment]
Hello. I've just noticed this intriguing app on MacUpdate and wondered if it is possible to use it with my VirginMedia ("VM") Tivo Box in the UK? Are there any special settings I need to allow a "VM" Tivo to connect, extract and convert media from the Tivo to my Mac HDD's running 10.10 (Yosemite)? I'd be most gratified if someone could answer this. If you require additional info; i.e., the Tivo model, HDMI Receiver set-up I have...? Anything, really.

Thanks (in anticipation).

**Comment by mackworth** [comment]
Should be fine.  I haven't heard of any problems there (as opposed to Australia).  Be sure to follow the [installation steps here](https://github.com/dscottbuch/cTiVo/wiki/Installation).  Once you've done that, you should be good.

**Comment by mackworth** [comment]
@UbiGit, did that work for you?

**Comment by WoodyO** [comment]
I might have missed a message. Did what work for me?

**Comment by mackworth** [comment]
@WoodyO, sorry; that was for the person known as @UbiGit!

**Comment by UbiGit** [comment]
@Mackworth I tried following the cTiVo set-up instructions but met with the message "You cannot activate service on this DVR. Please contact Virgin Media Customer Care by dialling 150 from your Virgin Media phone or 0845 454 1111, or visiting www.virginmedia.com/help."

There is also a dialogue box on the "TiVo Device Preferences" URL which states "Enable video downloads" but this has "N/A" beneath it.

I haven't contacted Virgin yet as I'm not sure what I should be asking of them. I presume they have to allow me to make changes from their end, or, change the settings themselves... Any suggestions?

**Comment by mackworth** [comment]
@UbiGit Well, I've done a little more research and it doesn't look good for Virgin Media. Basically you'd be asking them to enable the "TiVo to Go" functionality, specially that Enable Video Downloads. However, before you launch into that, my impression from some web searching is that although Virgin Media technically supports it, it marks all shows as "Copy Protected", which tells the TiVo to not allow the download.

**Comment by mackworth** [comment]
Okay, with the release of 2.4.0 today; I'm closing this chain. Thanks all for your help. If you have any problems, please open a new Issue!

**Comment by WoodyO** [comment]
Hugh,

It's been a pleasure. You've done a fantastic job here and I've enjoyed working with you. 

Simply as a matter in interest I am 86 years old and started my programming career with Woz’s Sweet Sixteen in 1979. Hell, I could do anything programming the 6502. Was having so much fun I opened my own computer store in 1980 with the Vector Graphic franchise in Concord, CA. We also sold KayPros and DECs. By then we had 8080 assembly, then Z80 and finally 8086 (16 bit -Wow!). When Vector came out with its 10Meg hard drive we had the world by the tail. Retired in 1990. Today I wouldn't know a flow chart from a hieroglyph.

WoodyO  ©¿©¬ 

===============================================================
